app/src/test/java/com/onboarding/calculator/CalculatorTest.java

```diff
@@ -0,0 +1,213 @@
+package com.onboarding.calculator;
+
+import com.onboarding.calculator.mvp.contract.CalculatorContract;
+import com.onboarding.calculator.mvp.model.CalculatorModel;
+import com.onboarding.calculator.mvp.presenter.CalculatorPresenter;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.onboarding.calculator.util.ConstantsUtils.ADD;
+import static com.onboarding.calculator.util.ConstantsUtils.DIV;
+import static com.onboarding.calculator.util.ConstantsUtils.EMPTY_STRING;
+import static com.onboarding.calculator.util.ConstantsUtils.Error;
+import static com.onboarding.calculator.util.ConstantsUtils.MUL;
+import static com.onboarding.calculator.util.ConstantsUtils.ONE_STRING_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_CLEAN_A_VALUE_IN_SECOND_OPERAND_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_CLEAN_OPERATOR_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_MINUS_IN_OPERATOR_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_MINUS_IN_SECOND_OPERAND_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.SUB;
+import static com.onboarding.calculator.util.ConstantsUtils.THREE_STRING_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.TWO_STRING_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.ZERO_STRING_TEST;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+public class CalculatorTest {
+
+    private CalculatorContract.Model model;
+    private CalculatorContract.Presenter presenter;
+    private CalculatorContract.View view;
+
+    @Before
+    public void setup() {
+        model = new CalculatorModel();
+        view = mock(CalculatorContract.View.class);
+        presenter = new CalculatorPresenter(model, view);
+    }
+
+    @Test
+    public void setACorrectOperator() {
+        //Test
+        presenter.onOperatorButtonPressed(ADD);
+        //Assert
+        assertEquals(ADD, model.getLastModified());
+        verify(view).showOperator(model.getLastModified());
+        verify(view).operationViewUpdate(model.getLastModified());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void setValuesInFirstOperand() {
+        //Test
+        presenter.onCalculatorButtonPressed(ONE_STRING_TEST);
+        //Assert
+        assertEquals(ONE_STRING_TEST, model.getLastModified());
```

Comment:
here, you can add another control if it is saved into the firstOperand no?

Reply:
Solved in a call!